lib/puppet/util/servicenow.rb

```diff
@@ -124,18 +124,30 @@ def human_readable_event_summary(resource_statuses)
   module_function :human_readable_event_summary
 
   def additional_info_resource_events(resource_statuses)
-    corrective_changes = []
-    intentional_changes = []
+    corrective_changes          = []
+    intentional_changes         = []
+    pending_corrective_changes  = []
+    pending_intentional_changes = []
+    failures                    = []
     resource_statuses.values.select { |resource| resource.out_of_sync == true || resource.failed == true }.each do |resource|
       event_summary = { 'resource'         => resource.resource,
                         'containing_class' => resource.resource_type,
                         'containment_path' => resource.containment_path,
                         'file'             => resource.file,
                         'line'             => resource.line }
-      (resource.corrective_change == true) ? (corrective_changes << event_summary) : (intentional_changes << event_summary)
+      if resource.failed
+        failures << event_summary
+      elsif resource.events.select { |event| event.status == 'noop' }.count > 0
+        (resource.corrective_change == true) ? (pending_corrective_changes << event_summary) : (pending_intentional_changes << event_summary)
+      else
+        (resource.corrective_change == true) ? (corrective_changes << event_summary) : (intentional_changes << event_summary)
+      end
     end
-    { 'corrective_changes'  => corrective_changes,
-      'intentional_changes' => intentional_changes }
+    { '__corrective_changes__'          => corrective_changes,
+      '__intentional_changes__'         => intentional_changes,
+      '__pending_corrective_changes__'  => pending_corrective_changes,
+      '__pending_intentional_changes__' => pending_intentional_changes,
+      '__failures__'                    => failures }
```

Comment:
So, I think this is my fault, but can we remove the underscores from these key names? I sent you screenshot of key names that had those underscores in the name, but that was purely for markdown formatting purposes. They don't actually need to be here in the real key name.

Reply:
ah cool, I was wondering if you were recommending this to prevent naming conflicts.

Reply:
Nope. Just careless with screenshots. :(